lib/ruby_lsp/tapioca/run_gem_rbi_check.rb

```diff
@@ -25,31 +25,43 @@ def initialize(project_path)
       def run
         return log_message("Not a git repository") unless git_repo?
 
-        lockfile_changed? ? generate_gem_rbis : cleanup_orphaned_rbis
+        if lockfile_changed?
+          generate_gem_rbis
+        else
+          cleanup_orphaned_rbis
+        end
       end
 
       private
 
       attr_reader :project_path
 
-      sig { returns(T::Boolean) }
+      sig { returns(T.nilable(T::Boolean)) }
```

Comment:
I still think this shouldn't return a `nil`. The call only needs to know if it's true or false. Feels like this is leaking an implementation detail of `Process::Status`.